integration/base/src/main/java/com/inrupt/client/integration/base/ApplicationRequestMetadataScenarios.java

```diff
@@ -0,0 +1,298 @@
+
+        // Create a new resource and check response headers
+        final Request.Builder reqBuilder = Request.newBuilder(resourceUri)
+                .header(Utils.CONTENT_TYPE, Utils.TEXT_TURTLE)
+                .PUT(Request.BodyPublishers.noBody());
+
+        //fill the configured headers
+        headers.forEach((key,value) -> {
+            reqBuilder.header(key, value.get(0));
+        });
+
+        final var req = reqBuilder.build();
+
+        final var res = authClient.send(req, Response.BodyHandlers.discarding());
+
+        assertTrue(Utils.isSuccessful(res.statusCode()));
+
+        for (String value: REQUEST_METADATA_HEADERS_THAT_PROPAGATE) {
+            assertEquals(headers.get(value), res.headers().allValues(value));
+        }
+    }
+
+    @ParameterizedTest
+    @MethodSource("provideSessions")
+    @DisplayName(" " +
+            "Request and response headers match for a successful authenticated request")
+    void requestResponseMatchOnAuthRequestAsyncHighLevelClientTest(final Session session) {
+        assumeTrue(featureIsActive());
+
+        LOGGER.info("Integration Test - High level async client -" +
+                " Request and response headers match for a successful authenticated request");
+
+        final var headers = fillHeaders("bbbbbb");
+        headers.put("someblabla", List.of("bbbbbb-2454-4501-8110-ecc082aa975f"));
+
+        final SolidClient authClient = SolidClient.getClientBuilder()
+                        .headers(Headers.of(headers)).build()
+                        .session(session);
+
+        final String resourceName = privateContainerURI + "e2e-test-application-metadata-" + UUID.randomUUID();
+        final String predicateName = "https://example.example/predicate";
+        final IRI booleanType = rdf.createIRI("http://www.w3.org/2001/XMLSchema#boolean");
+
+        final IRI subject = rdf.createIRI(resourceName);
+        final IRI predicate = rdf.createIRI(predicateName);
+        final Literal objectTrue = rdf.createLiteral("true", booleanType);
+
+        // Populate data for this resource
+        final Dataset dataset = rdf.createDataset();
+        dataset.add(null, subject, predicate, objectTrue);
+
+        // Create a new resource and check response headers
+        final URI resourceUri = URI.create(resourceName);
+        authClient.create(new SolidRDFSource(resourceUri, dataset)).thenAccept(response -> {
+            for (String value: REQUEST_METADATA_HEADERS_THAT_PROPAGATE) {
+                assertEquals(headers.get(value), response.getHeaders().allValues(value));
+            }
+            assertTrue(response.getHeaders().allValues("someblabla").isEmpty());
+        });
+    }
+
+    @ParameterizedTest
+    @MethodSource("provideSessions")
+    @DisplayName(" " +
+            "Request and response headers match for a successful authenticated request")
+    void requestResponseMatchOnAuthRequestHighLevelSyncClientTest(final Session session) {
```

Comment:
The prefixes aren't in the same order as the previous test: should this be `requestResponseMatchOnAuthRequestSyncHighLevelClientTest`, or the other one be `requestResponseMatchOnAuthRequestHighLevelAsyncClientTest`?